src/mango/src/mango_cursor_text.erl

```diff
@@ -108,7 +109,8 @@ execute(Cursor, UserFun, UserAcc) ->
         user_fun = UserFun,
         user_acc = UserAcc,
         fields = Cursor#cursor.fields,
-        execution_stats = mango_execution_stats:log_start(Stats)
+        execution_stats = mango_execution_stats:log_start(Stats),
+        documents_seen = #{}
```

Comment:
Could use a set probably. Use version = 2 as it will be backed by a map anyway.

https://www.erlang.org/docs/24/man/sets#new-1

Reply:
I was considering that too.  But then I decided to mimic the implementation of [`lists:uniq/1`](https://github.com/erlang/otp/blob/1d47ecb040b90a8011a200335400279dcb745bb2/lib/stdlib/src/lists.erl#L3222-L3240) since we basically need to weave its definition into the streaming as there is no accumulated list there to run on that.

Reply:
Could just check the set seen documents and emit it if it's not in the set and add it, and if it is in the set skip it?

Reply:
That is what I am doing here but with a map directly?

Reply:
I guess I got confused by the reference to lists:uniq/1 as it implies we're doing some lists processing or some other complicated operation. If we need a set data structure only, why not use the built-in set module?

Reply:
version 2 set are using maps too so it should be just as efficient, for a 200 max element set it might be better to opt for code readability instead